spec/security.md

```diff
@@ -1,4 +1,64 @@
 Security Considerations {#sec-security}
 =====================
 
-**TODO**(shaseley): Fill this in.
+<div class="non-normative">
+*This section is non-normative.*
+
+The main security consideration for the APIs defined in this specification is
+whether or not any information is potentially leaked between origins by
+timing-based side-channel attacks.
+</div>
+
+`postTask` as a High-Resolution Timing Source {#sec-security-high-res-timer}
+---------------------
+
+<div class="non-normative">
+The first consideration is whether the {{Scheduler/postTask()}} API can be used
+as a high-resolution timing source. {{Scheduler/postTask()}}'s
+{{SchedulerPostTaskOptions/delay}}, like
+{{WindowOrWorkerGlobalScope/setTimeout()}}'s timeout value, is expressed in
+whole milliseconds (the minimum non-zero delay being 1 ms), and there is no
+guarantee that tasks will run exactly when the delay expires since tasks are
+queued when the delay expires. Given this, we do not believe the API can be
+used as a high-resolution timing source, but we mention it here because of the
+general interest in the topic.
+</div>
+
+Monitoring Another Origin's Tasks {#sec-security-monitoring-tasks}
+---------------------
+
+<div class="non-normative">
+The second consideration is whether {{Scheduler/postTask()}} leaks any
+information about other origins' tasks. The threat model we consider is code
+from two different origins running in separate event loops in the same thread.
+
+Because the UA can only run tasks from one event loop at a time, an attacker
+might be able to gain information about tasks running in another event loop by
+monitoring when their tasks run. For example, an attacker could flood the
+system with {{TaskPriority/user-blocking}} tasks and expect them to run
+consecutively; if there are large gaps in between, then the attacker might
+infer something about tasks running in another event loop, e.g. they have
+higher priority.
+
+The first thing we note is that the attacker would not be able to definitively
+tell that a task ran in another event loop, e.g. an internal browser task might
+have run, or perhaps the UA throttled the attacker's script.  Second, any
+information gained would be implementation-dependent since inter-event-loop
```

Comment:
I'm not sure it's really a defense that the attacker would have to implement their attack for each of ~3 implementations.

Reply:
Maybe "The information exposed would depend on implementation details, and implementations can reduce it as described below."